src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarEntry.cs

```diff
@@ -138,10 +138,12 @@ public string LinkName
         /// <remarks>The value in this field has no effect on Windows platforms.</remarks>
         public UnixFileMode Mode
         {
-            get => (UnixFileMode)_header._mode;
+            // Some paths do not use the setter, and we want to return valid UnixFileMode.
+            // This mask only keeps the least significant 12 bits.
+            get => (UnixFileMode)(_header._mode & (int)TarHelpers.ValidUnixFileModes);
             set
             {
-                if ((int)value is < 0 or > 4095) // 4095 in decimal is 7777 in octal
+                if ((value & ~TarHelpers.ValidUnixFileModes) != 0) // throw on invalid UnixFileModes
```

Comment:
nit: There's another reference to `4095` on the tests which we could change to a `const UnixFileMode ValidUnixFileModes`.
https://github.com/dotnet/runtime/blob/ccd9d169b0189d0503ec885bf86e9f026cb621a8/src/libraries/System.Formats.Tar/tests/TarWriter/TarWriter.File.Base.Unix.cs#L38

Reply:
Currently we don't have access to the helpers in the tests.

Reply:
I was suggesting _copying_ the `const` to the tests, not making the one on src accessible to tests. Is just a nit, so you can disregard it.